File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py

num=int(input())
input_list=[]
for i in range(num):
  n=int(input())
  input_list.append(n)
result=[]
for i in range(num):
  tmp_list=[1,2,4]
  n=input_list[i]
  if n==1 or n==2 or n==3:
    result.append(tmp_list[n-1])
    continue;
  for j in range(3,n):
    tmp_list.append(tmp_list[j-1]+tmp_list[j-2]+tmp_list[j-3])
  result.append(tmp_list[-1])
for i in range(num):
  print(result[i])

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.js
const fs=require('fs');
let input=fs.readFileSync(0).toString().trim().split('\n');
const n=+input[0]
let list=[];

for(let i=1;i<=n;i++){
    let a=+input[i];
    list.push(a);
}

for(let i=0;i<n;i++){
    let result=[1,2,4];
    let target=list[i]
    for(let j=3;j<target;j++){
        result.push(result[j-1]+result[j-2]+result[j-3]);
    }
    target===3||target===1||target===2?console.log(result[target-1]):console.log(result[target-1])
}
